### PR TITLE
Wait Module Preperation

### PR DESCRIPTION
**Pull request by harvey0100**
Preperation of Wait Module to AAutils - Updated Docstrings - Improved Unit Tests / Added Functional Tests.

## Summary by CodeRabbit

* **Documentation**
  * Expanded wait utility docs with detailed behavior, timing semantics, examples, and usage guidance.

* **Tests**
  * Added comprehensive unit and functional tests covering timing, retries, edge cases, exception propagation, and real I/O/threading scenarios.
  * Updated selftest totals to reflect new test counts.

* **Chores**
  * Added entries to the spell-ignore list.

### CODE REVIEW

File: selftests/functional/utils/wait.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+import socket
+import tempfile
+import threading
+import time
+import unittest
+
+from avocado.utils import wait
+
+
+class WaitForFunctionalTest(unittest.TestCase):
```
Comment (richtja) [inline selftests/functional/utils/wait.py]: We can use `TestCaseTmpDir` class as parent for this class to deal with tmp files preparations.

File: selftests/functional/utils/wait.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
+import os
```
Comment (richtja) [inline selftests/functional/utils/wait.py:1]: IMO we don't need that many tests for one method in `wait` utility. Most of these tests the same thing over and over again. For example, IMO from `wait_for` method point of view there is no difference between `test_file_appears` and `test_environment_variable_appears`.

File: selftests/functional/utils/wait.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        # Wait for a file that will never be created
+        start = time.time()
+        result = wait.wait_for(lambda: os.path.exists(filepath), timeout=0.5, step=0.1)
+        elapsed = time.time() - start
+
+        self.assertIsNone(result)
+        self.assertGreaterEqual(elapsed, 0.5)
+        self.assertLess(elapsed, 0.7)
+
+    def test_complex_condition_with_error_handling(self):
```
Comment (richtja) [inline selftests/functional/utils/wait.py]: Can you please explain what is the reason for this test?  If the purpose is to test what will happen when the condition raise error, then IMO it's not working properly because the error won't be raised. If the test just adds a more complicated condition, then I don't see a difference between this and `test_condition_becomes_true`, but maybe I am missing something.

Comment (harvey0100) [reply selftests/functional/utils/wait.py]: Hi Jan, I took a look into this again and it was a test that the AI side added.
I'm in complete agreement with you here and is a redundant test that is already being replicated i have removed it in the latest push.

Comment (harvey0100) [reply selftests/functional/utils/wait.py]: The Autils Side has also been updated removing this test.

File: selftests/functional/utils/wait.py (hunk 2 above)
Comment (clebergnu) [reply selftests/functional/utils/wait.py:1]: Based on what I saw here and in the `process` tests, the AI in use can be a bit overwhelming and repetitive when it comes to creating tests.  Based on the overall coverage results, all these tests increased the *project* coverage by 0.01% percent.  It'd be interesting to see the coverage change (percent wise) in the actual module.  Also, IMO there are two factors to judge whether these tests should be added:
1. The amount of coverage they add
2. How easy they make to spot a regression, that is, while one test can cover a lot of ground (say a functional test), unit tests will be very useful at showing the exact regression that was introduced.
